src/gadgets/utils.rs

```diff
@@ -200,6 +200,48 @@ pub fn alloc_num_equals<F: PrimeField, CS: ConstraintSystem<F>>(
   Ok(r)
 }
 
+/// Check that two numbers are equal and return a bit
+pub fn alloc_num_equals_const<F: PrimeField, CS: ConstraintSystem<F>>(
+  mut cs: CS,
+  a: &AllocatedNum<F>,
+  b: F,
+) -> Result<AllocatedBit, SynthesisError> {
+  // Allocate and constrain `r`: result boolean bit.
+  // It equals `true` if `a` equals `b`, `false` otherwise
+  let r_value = match (a.get_value(), b) {
+    (Some(a), b) => Some(a == b),
+    _ => None,
+  };
+
+  let r = AllocatedBit::alloc(cs.namespace(|| "r"), r_value)?;
+
+  // Allocate t s.t. t=1 if z1 == z2 else 1/(z1 - z2)
+
+  let t = AllocatedNum::alloc(cs.namespace(|| "t"), || {
+    Ok(if *a.get_value().get()? == b {
+      F::ONE
+    } else {
+      (*a.get_value().get()? - b).invert().unwrap()
+    })
+  })?;
```

Comment:
In situations where we want to compute a boolean selector vector, we may want to leverage `batch_invert` when computing the `t`s for each bit. Inversion is an expensive operation, and the threshold at which `batch_invert` becomes cheaper than individual inversions is very low (each additional element to the batch is only 3 field multiplications.